File: src/main/java/hzt/aoc/day13/Day13Challenge.java
package hzt.aoc.day13;

import hzt.aoc.Challenge;

public abstract class Day13Challenge extends Challenge {

    Day13Challenge(String challengeTitle, String description) {
        super(challengeTitle, description, "20201213-input-day13.txt");
    }

    Day13Challenge(String challengeTitle, String description, String inputFileName) {
        super(challengeTitle, description, inputFileName);
    }



    abstract String getMessage(Number value);
}

File: src/main/kotlin/hzt/aoc/day13/Day13Challenge.kt
package hzt.aoc.day13

import hzt.aoc.Challenge

abstract class Day13Challenge : Challenge {
    internal constructor(challengeTitle: String, description: String) : super(
        challengeTitle,
        description,
        "20201213-input-day13.txt"
    )

    internal constructor(challengeTitle: String, description: String, inputFileName: String) : super(
        challengeTitle,
        description,
        inputFileName
    )
}
